litestar/typing.py

```diff
@@ -35,6 +35,19 @@
 T = TypeVar("T", bound=KwargDefinition)
 
 
+class _KwargMetaExtractor(Protocol):
+    @staticmethod
+    def matches(annotation: Any, name: str | None, default: Any) -> bool:
+        ...
+
+    @staticmethod
+    def extract(annotation: Any, default: Any) -> Any:
+        ...
+
+
+_KWARG_META_EXTRACTORS: set[_KwargMetaExtractor] = set()
```

Comment:
This is not a long term solution, but the only somewhat sane way of doing this at the moment. I really don't like using global variables like this, but we were relying on global state before as well, it was just hidden behind the pydantic utility functions. This at least makes it visible and decouples the core utilities from the pydantic utils.